<html><head><title>References Cited in the Manual of ape</title></head>
<body bgcolor="lightyellow">
<h2>References Cited in the Manual of ape</h2>

<p><a href="http://archive.numdam.org/ARCHIVE/RO/RO_2001__35_2/RO_2001__35_2_283_0/RO_2001__35_2_283_0.pdf">http://archive.numdam.org/ARCHIVE/RO/RO_2001__35_2/RO_2001__35_2_283_0/RO_2001__35_2_283_0.pdf</a> </p>
<p><a href="http://www.biomedcentral.com/1471-2105/9/166">http://www.biomedcentral.com/1471-2105/9/166</a> </p>
<p>Aldous, D. J. (2001) Stochastic models and descriptive statistics for phylogenetic trees, from Yule to today. <EM>Statistical Science</EM>, <B>16</B>, 23&ndash;34.
<p>Anderson, M. J. and Legendre, P. (1999) An empirical comparison of permutation methods for tests of partial regression coefficients in a linear model. <EM>Journal of Statistical Computation and Simulation</EM>, <B>62</B>, 271&ndash;303. </p>
<p>Anonymous. FASTA format description. <a href="http://www.ncbi.nlm.nih.gov/BLAST/fasta.html">http://www.ncbi.nlm.nih.gov/BLAST/fasta.html</a> </p>
<p>Anonymous. IUPAC ambiguity codes. <a href="http://www.ncbi.nlm.nih.gov/SNP/iupac.html">http://www.ncbi.nlm.nih.gov/SNP/iupac.html</a> </p>
<p>Barraclough, T. G., Harvey, P. H. and  Nee, S. (1995) Sexual selection and taxonomic diversity in passerine birds. <EM>Proceedings of the Royal Society of London. Series B. Biological Sciences</EM>, <B>259</B>, 211&ndash;215. </p>
<p>Barraclough, T. G., Harvey, P. H., and Nee,  S. (1996) Rate of <EM>rbc</EM>L gene sequence evolution and species diversification in flowering plants (angiosperms). <EM>Proceedings of the Royal Society of London. Series B. Biological Sciences</EM>, <B>263</B>, 589&ndash;591. </p>
<p>Barry, D. and Hartigan, J. A. (1987) Asynchronous distance between homologous DNA sequences. <EM>Biometrics</EM>, <B>43</B>, 261&ndash;276. </p>
<p>Billera, L. J., Holmes, S. P. and Vogtmann, K. (2001) Geometry of the space of phylogenetic trees. <EM>Advances in Applied Mathematics</EM>, <B>27</B>, 733&ndash;767. </p>
<p>Blomberg, S. P., Garland, Jr, T., and Ives, A. R. (2003) Testing for phylogenetic signal in comparative data: behavioral traits are more labile. <EM>Evolution</EM>, <B>57</B>, 717&ndash;745.
<p>Britton, T., Oxelman, B., Vinnersten, A. and Bremer, K. (2002) Phylogenetic dating with confidence intervals using mean path lengths. <EM>Molecular Phylogenetics and Evolution</EM>, <B>24</B>, 58&ndash;65. </p>
<p>Cailliez, F. (1983) The analytical solution of the additive constant problem. <EM>Psychometrika</EM>, <B>48</B>, 305&ndash;308. </p>
<p>Campbell, V., Legendre, P. and Lapointe, F.-J. (2009) Assessing congruence among ultrametric distance matrices. <EM>Journal of Classification</EM>, <B>26</B>, 103&ndash;117. </p>
<p>Campbell, V., Legendre, P. and Lapointe, F.-J. (2011) The performance of the Congruence Among Distance Matrices (CADM) test in phylogenetic analysis. <EM>BMC Evolutionary Biology</EM>, <B>11</B>, 64. <a href="http://www.biomedcentral.com/1471-2148/11/64">http://www.biomedcentral.com/1471-2148/11/64</a>. </p>
<p>Chenna, R., Sugawara, H., Koike, T., Lopez, R., Gibson, T. J., Higgins, D. G. and Thompson, J. D. (2003) Multiple sequence alignment with the Clustal series of programs. <EM>Nucleic Acids Research</EM> <B>31</B>, 3497&ndash;3500. <a href="http://www.clustal.org/">http://www.clustal.org/</a> </p>
<p>Cheverud, J. M., Dow, M. M. and Leutenegger, W. (1985) The quantitative assessment of phylogenetic constraints in comparative analyses: sexual dimorphism in body weight among primates. <EM>Evolution</EM>, <B>39</B>, 1335&ndash;1351. </p>
<p>Criscuolo, A. and Gascuel, O. (2008). Fast NJ-like algorithms to deal with incomplete distance matrices. <EM>BMC Bioinformatics</EM>, 9. </p>
<p>Criscuolo, A., Berry, V., Douzery, E. J. P. , and Gascuel, O. (2006) SDM: A fast distance-based approach for (super)tree building in phylogenomics. <EM>Systematic Biology</EM>, <B>55</B>, 740&ndash;755. </p>
<p>Cunningham, C. W., Omland, K. E. and Oakley, T. H. (1998) Reconstructing ancestral character states: a critical reappraisal. <EM>Trends in Ecology &amp; Evolution</EM>, <B>13</B>, 361&ndash;366. </p>
<p>Desper, R. and Gascuel, O. (2002) Fast and accurate phylogeny reconstruction algorithms based on the minimum-evolution principle. <EM>Journal of Computational Biology</EM>, <B>9(5)</B>, 687&ndash;705. </p>
<p>Diaconis, P. W. and Holmes, S. P. (1998) Matchings and phylogenetic trees. <EM>Proceedings of the National Academy of Sciences USA</EM>, <B>95</B>, 14600&ndash;14602. </p>
<p>Edgar, R. C. (2004) MUSCLE: Multiple sequence alignment with high accuracy and high throughput. <EM>Nucleic Acids Research</EM>, <B>32</B>, 1792&ndash;1797. <a href="http://www.drive5.com/muscle/muscle_userguide3.8.html">http://www.drive5.com/muscle/muscle_userguide3.8.html</a> </p>
<p>Efron, B., Halloran, E. and Holmes, S. (1996) Bootstrap confidence levels for phylogenetic trees. <EM>Proceedings of the National Academy of Sciences USA</EM>, <B>93</B>, 13429&ndash;13434. </p>
<p>Farris, J. M. (1970) Methods for computing Wagner trees. <EM>Systematic Zoology</EM>, <B>19</B>, 83&ndash;92. </p>
<p>Felsenstein, J. (1981) Evolutionary trees from DNA sequences: a maximum likelihood approach. <EM>Journal of Molecular Evolution</EM>, <B>17</B>, 368&ndash;376. </p>
<p>Felsenstein, J. (1985) Confidence limits on phylogenies: an approach using the bootstrap. <EM>Evolution</EM>, <B>39</B>, 783&ndash;791. </p>
<p>Felsenstein, J. (1985) Phylogenies and the comparative method. <EM>American Naturalist</EM>, <B>125</B>, 1&ndash;15. </p>
<p>Felsenstein, J. (1993) Phylip (Phylogeny Inference Package) version 3.5c. Department of Genetics, University of Washington. <a href="http://evolution.genetics.washington.edu/phylip/phylip.html">http://evolution.genetics.washington.edu/phylip/phylip.html</a> </p>
<p>Felsenstein, J. (2004) <EM>Inferring phylogenies</EM>. Sunderland: Sinauer Associates. </p>
<p>Felsenstein, J. (2004) Phylip (Phylogeny Inference Package) version 3.68. Department of Genetics, University of Washington, Seattle, USA. <a href="http://evolution.genetics.washington.edu/phylip/phylip.html">http://evolution.genetics.washington.edu/phylip/phylip.html</a>. </p>
<p>Felsenstein, J. (2008) Comparative methods with sampling error and within-species variation: Contrasts revisited and revised. <EM>American Naturalist</EM>, <B>171</B>, 713&ndash;725. </p>
<p>Felsenstein, J. and Churchill, G. A. (1996) A Hidden Markov model approach to variation among sites in rate of evolution. <EM>Molecular Biology and Evolution</EM>, <B>13</B>, 93&ndash;104. </p>
<p>Felsenstein, J. The Newick tree format. <a href="http://evolution.genetics.washington.edu/phylip/newicktree.html">http://evolution.genetics.washington.edu/phylip/newicktree.html</a> </p>
<p>Freckleton, R. P., Harvey, P. H. and M. Pagel, M. (2002) Phylogenetic analysis and comparative data: a test and review of evidence. <EM>American Naturalist</EM>, <B>160</B>, 712&ndash;726. </p>
<p>Friedman, M. (1937) The use of ranks to avoid the assumption of normality implicit in the analysis of variance. <EM>Journal of the American Statistical Association</EM>, <B>32</B>, 675&ndash;701. </p>
<p>Galtier, N. and Gouy, M. (1995) Inferring phylogenies from DNA sequences of unequal base compositions. <EM>Proceedings of the National Academy of Sciences USA</EM>, <B>92</B>, 11317&ndash;11321. </p>
<p>Garland, T. Jr. and Ives, A. R. (2000) Using the past to predict the present: confidence intervals for regression equations in phylogenetic comparative methods. <EM>American Naturalist</EM>, <B>155</B>, 346&ndash;364. </p>
<p>Gascuel, O. (1997) BIONJ: an improved version of the NJ algorithm based on a simple model of sequence data. <EM>Molecular Biology and Evolution</EM>, <B>14:</B>, 685&ndash;695. </p>
<p>Gascuel, O. (2000). Data model and classification by trees: the minimum variance reduction (MVR) method. <EM>Journal of Classification</EM>, <B>17</B>, 67&ndash;99. </p>
<p>Gillespie, D. T. (1996) Exact numerical simulation of the Ornstein-Uhlenbeck process and its integral. <EM>Physical Review E</EM>, <B>54</B>, 2084&ndash;2091. </p>
<p>Gittleman, J. L. and Kot, M. (1990) Adaptation: statistics and a null model for estimating phylogenetic effects. <EM>Systematic Zoology</EM>, <B>39</B>, 227&ndash;241. </p>
<p>Gower, J. C. (1966) Some distance properties of latent root and vector methods used in multivariate analysis. <EM>Biometrika</EM>, <B>53</B>, 325&ndash;338. </p>
<p>Gower, J. C. and Legendre, P. (1986) Metric and Euclidean properties of dissimilarity coefficients. <EM>Journal of Classification</EM>, <B>3</B>, 5&ndash;48. </p>
<p>Grafen, A. (1989) The phylogenetic regression. <EM>Philosophical Transactions of the Royal society of London. Series B. Biological Sciences</EM>, <B>326</B>, 119&ndash;157. </p>
<p>Guindon, S. and Gascuel, O. (2003) A simple, fast, and accurate algorithm to estimate large phylogenies by maximum likelihood. <EM>Systematic Biology</EM>, <B>52</B>, 696&ndash;704. <a href="http://www.atgc-montpellier.fr/phyml/">http://www.atgc-montpellier.fr/phyml/</a> </p>
<p>Gu, X. and Li, W.-H. (1996) Bias-corrected paralinear and LogDet distances and tests of molecular clocks and phylogenies under nonstationary nucleotide frequencies. <EM>Molecular Biology and Evolution</EM>, <B>13</B>, 1375&ndash;1383. </p>
<p>Hafner, M. S, P. D. Sudman, F. X. Villablanca, T. A. Spradling, J. W. Demastes and S. A. Nadler. 1994. Disparate rates of molecular evolution in cospeciating hosts and parasites. <EM>Science</EM>, <B>265</B>, 1087&ndash;1090. </p>
<p>Hanazawa, M., Narushima, H. and Minaka, N. (1995) Generating most parsimonious reconstructions on a tree: a generalization of the Farris&ndash;Swofford&ndash;Maddison method. <EM>Discrete Applied Mathematics</EM>, <B>56</B>, 245&ndash;265. </p>
<p>Hansen, T. F. (1997) Stabilizing selection and the comparative analysis of adaptation. <EM>Evolution</EM>, <B>51</B>, 1341&ndash;1351. </p>
<p>Harvey, P. H. and Pagel, M. D. (1991) <EM>The comparative method in evolutionary biology</EM>. Oxford University Press. </p>
<p>Harvey, P. H., May, R. M. and Nee, S. (1994) Phylogenies without fossils. <EM>Evolution</EM>, <B>48</B>, 523&ndash;529. </p>
<p>Holland, B. R., Huber, K. T., Dress, A. and Moulton, V. (2002) Delta plots: a tool for analyzing phylogenetic distance data. <EM>Molecular Biology and Evolution</EM>, <B>12</B>, 2051&ndash;2059. </p>
<p>Hubert, N., Paradis, E., Bruggemann, H. and Planes, S. (2011) Community assembly and diversification in Indo-Pacific coral reef fishes. <EM>Ecology and Evolution</EM>, <B>1</B>, 229&ndash;277. </p>
<p>Jin, L. and Nei, M. (1990) Limitations of the evolutionary parsimony method of phylogenetic analysis. <EM>Molecular Biology and Evolution</EM>, <B>7</B>, 82&ndash;102. </p>
<p>Jukes, T. H. and Cantor, C. R. (1969) Evolution of protein molecules. in <EM>Mammalian Protein Metabolism</EM>, ed. Munro, H. N., pp. 21&ndash;132, New York: Academic Press. </p>
<p>Kendall, D. G. (1948) On the generalized &ldquo;birth-and-death&rdquo; process. <EM>Annals of Mathematical Statistics</EM>, <B>19</B>, 1&ndash;15. </p>
<p>Kendall, M. G. and Babington Smith, B. (1939) The problem of m rankings. <EM>Annals of Mathematical Statistics</EM>, <B>10</B>, 275&ndash;287. </p>
<p>Kimura, M. (1980) A simple method for estimating evolutionary rates of base substitutions through comparative studies of nucleotide sequences. <EM>Journal of Molecular Evolution</EM>, <B>16</B>, 111&ndash;120. </p>
<p>Kimura, M. (1981) Estimation of evolutionary distances between homologous nucleotide sequences. <EM>Proceedings of the National Academy of Sciences USA</EM>, <B>78</B>, 454&ndash;458. </p>
<p>Kuhner, M. K. and Felsenstein, J. (1994) Simulation comparison of phylogeny algorithms under equal and unequal evolutionary rates. <EM>Molecular Biology and Evolution</EM>, <B>11</B>, 459&ndash;468. </p>
<p>Lake, J. A. (1994) Reconstructing evolutionary trees from DNA and protein sequences: paralinear distances. <EM>Proceedings of the National Academy of Sciences USA</EM>, <B>91</B>, 1455&ndash;1459. </p>
<p>Lapointe, F.-J., Kirsch, J. A. W. and Hutcheon, J. M. (1999) Total evidence, consensus, and bat phylogeny: a distance-based approach. <EM>Molecular Phylogenetics and Evolution</EM>, <B>11</B>, 55&ndash;66. </p>
<p>Legendre, P. (2010) Coefficient of concordance. Pp. 164-169 in: Encyclopedia of Research Design, Vol. 1. N. J. Salkind, ed. SAGE Publications, Inc., Los Angeles. </p>
<p>Legendre, P. and Desdevises, Y. (2009) Independent contrasts and regression through the origin. <EM>Journal of Theoretical Biology</EM>, <B>259</B>, 727&ndash;743. </p>
<p>Legendre, P. and Gallagher, E. D. (2001) Ecologically meaningful transformations for ordination of species data. <EM>Oecologia</EM>, <B>129</B>, 271&ndash;280. </p>
<p>Legendre, P. and Lapointe, F.-J. (2004) Assessing congruence among distance matrices: single malt Scotch whiskies revisited. <EM>Australian and New Zealand Journal of Statistics</EM>, <B>46</B>, 615&ndash;629. </p>
<p>Legendre, P. and Lapointe, F.-J. (2005) Congruence entre matrices de distance. P. 178-181 in: Makarenkov, V., G. Cucumel et F.-J. Lapointe [eds] Comptes rendus des 12emes Rencontres de la Societe Francophone de Classification, Montreal, 30 mai - 1er juin 2005. </p>
<p>Legendre, P. and Legendre, L. (1998) <EM>Numerical ecology, 2nd English edition.</EM> Amsterdam: Elsevier Science BV. </p>
<p>Legendre, P., Y. Desdevises and E. Bazin. 2002. A statistical test for host-parasite coevolution. <EM>Systematic Biology</EM>, <B>51(2)</B>, 217&ndash;234. </p>
<p>Lingoes, J. C. (1971) Some boundary conditions for a monotone analysis of symmetric matrices. <EM>Psychometrika</EM>, <B>36</B>, 195&ndash;203. </p>
<p>Liu, L., Yu, L. and Pearl, D. K. (2010) Maximum tree: a consistent estimator of the species tree. <EM>Journal of Mathematical Biology</EM>, <B>60</B>, 95&ndash;106. </p>
<p>Lockhart, P. J., Steel, M. A., Hendy, M. D. and Penny, D. (1994) Recovering evolutionary trees under a more realistic model of sequence evolution. <EM>Molecular Biology and Evolution</EM>, <B>11</B>, 605&ndash;602. </p>
<p>Lynch, M. (1991) Methods for the analysis of comparative data in evolutionary biology. <EM>Evolution</EM>, <B>45</B>, 1065&ndash;1080. </p>
<p>Maddison, D. R., Swofford, D. L. and Maddison, W. P. (1997) NEXUS: an extensible file format for systematic information. <EM>Systematic Biology</EM>, <B>46</B>, 590&ndash;621. </p>
<p>Maddison, W. P. and Knowles, L. L. (2006) Inferring phylogeny despite incomplete lineage sorting. <EM>Systematic Biology</EM>, <B>55</B>, 21&ndash;30. </p>
<p>Makarenkov, V. and Lapointe, F.-J. (2004) A weighted least-squares approach for inferring phylogenies from incomplete distance matrices. <EM>Bioinformatics</EM>, <B>20</B>, 2113&ndash;2121. <a href="http://dx.doi.org/10.1093/bioinformatics/bth211">http://dx.doi.org/10.1093/bioinformatics/bth211</a>
<p>Manly, B. F. J. (1986) <EM>Multivariate statistical methods: a primer.</EM> London: Chapman &amp; Hall. </p>
<p>Mantel, N. (1967) The detection of disease clustering and a generalized regression approach. <EM>Cancer Research</EM>, <B>27</B>, 209&ndash;220. </p>
<p>Martins, E. P. and Hansen, T. F. (1997) Phylogenies and the comparative method: a general approach to incorporating phylogenetic information into the analysis of interspecific data. <EM>American Naturalist</EM>, <B>149</B>, 646&ndash;667. </p>
<p>McConway, K. J. and Sims, H. J. (2004) A likelihood-based method for testing for nonstochastic variation of diversification rates in phylogenies. <EM>Evolution</EM>, <B>58</B>, 12&ndash;23. </p>
<p>McGuire, G., Prentice, M. J. and Wright, F. (1999). Improved error bounds for genetic distances from DNA sequences. <EM>Biometrics</EM>, <B>55</B>, 1064&ndash;1070. </p>
<p>McKenzie, A. and Steel, M. (2000) Distributions of cherries for two models of trees. <EM>Mathematical Biosciences</EM>, <B>164</B>, 81&ndash;92. </p>
<p>Misawa, K. (2000) A simple method for classifying genes and a bootstrap test for classifications. <EM>Molecular Biology and Evolution</EM>, <B>17</B>, 1879&ndash;1884. </p>
<p>Narushima, H. and Hanazawa, M. (1997) A more efficient algorithm for MPR problems in phylogeny. <EM>Discrete Applied Mathematics</EM>, <B>80</B>, 231&ndash;238. </p>
<p>Nee, S., Holmes, E. C., May, R. M. and Harvey, P. H. (1995) Estimating extinctions from molecular phylogenies. in <EM>Extinction Rates</EM>, eds. Lawton, J. H. and May, R. M., pp. 164&ndash;182, Oxford University Press. </p>
<p>Nee, S., Holmes, E. C., Rambaut, A. and Harvey, P. H. (1995) Inferring population history from molecular phylogenies. <EM>Philosophical Transactions of the Royal Society of London. Series B. Biological Sciences</EM>, <B>349</B>, 25&ndash;31. </p>
<p>Nee, S., May, R. M. and Harvey, P. H. (1994) The reconstructed evolutionary process. <EM>Philosophical Transactions of the Royal Society of London. Series B. Biological Sciences</EM>, <B>344</B>, 305&ndash;311. </p>
<p>Nei, M. and Kumar, S. (2000) <EM>Molecular evolution and phylogenetics</EM>. Oxford: Oxford University Press. </p>
<p>Notredame, C., Higgins, D. and Heringa, J. (2000) T-Coffee: A novel method for multiple sequence alignments. <EM>Journal of Molecular Biology</EM>, <B>302</B>, 205&ndash;217. <a href="http://www.tcoffee.org/Documentation/t_coffee/t_coffee_technical.htm">http://www.tcoffee.org/Documentation/t_coffee/t_coffee_technical.htm</a> </p>
<p>Olsen, G. Interpretation of the &quot;Newick's 8:45&quot; tree format standard. <a href="http://evolution.genetics.washington.edu/phylip/newick_doc.html">http://evolution.genetics.washington.edu/phylip/newick_doc.html</a> </p>
<p>Opgen-Rhein, R., Fahrmeir, L. and Strimmer, K. 2005. Inference of demographic history from genealogical trees using reversible jump Markov chain Monte Carlo. <EM>BMC Evolutionary Biology</EM>, <B>5</B>, 6. </p>
<p>Pagel, M. (1994) Detecting correlated evolution on phylogenies: a general method for the comparative analysis of discrete characters. <EM>Proceedings of the Royal Society of London. Series B. Biological Sciences</EM>, <B>255</B>, 37&ndash;45. </p>
<p>Pagel, M. (1999) Inferring the historical patterns of biological evolution. <EM>Nature</EM>, <B>401</B>,877&ndash;884.
<p>Pan, W. (2001) Akaike's information criterion in generalized estimating equations. <EM>Biometrics</EM>, <B>57</B>, 120&ndash;125. </p>
<p>Paradis, E. (1997) Assessing temporal variations in diversification rates from phylogenies: estimation and hypothesis testing. <EM>Proceedings of the Royal Society of London. Series B. Biological Sciences</EM>, <B>264</B>, 1141&ndash;1147. </p>
<p>Paradis, E. (1998) Testing for constant diversification rates using molecular phylogenies: a general approach based on statistical tests for goodness of fit. <EM>Molecular Biology and Evolution</EM>, <B>15</B>, 476&ndash;479. </p>
<p>Paradis, E. (2003) Analysis of diversification: combining phylogenetic and taxonomic data. <EM>Proceedings of the Royal Society of London. Series B. Biological Sciences</EM>, <B>270</B>, 2499&ndash;2505. </p>
<p>Paradis, E. (2005) Statistical analysis of diversification with species traits. <EM>Evolution</EM>, <B>59</B>, 1&ndash;12. </p>
<p>Paradis, E. (2006) <EM>Analyses of Phylogenetics and Evolution with R.</EM> New York: Springer. </p>
<p>Paradis, E. (2007) A Bit-Level Coding Scheme for Nucleotides. <a href="http://ape.mpl.ird.fr/misc/BitLevelCodingScheme_20April2007.pdf">http://ape.mpl.ird.fr/misc/BitLevelCodingScheme_20April2007.pdf</a> </p>
<p>Paradis, E. (2008) Definition of Formats for Coding Phylogenetic Trees in R. <a href="http://ape.mpl.ird.fr/misc/FormatTreeR_28July2008.pdf">http://ape.mpl.ird.fr/misc/FormatTreeR_28July2008.pdf</a> </p>
<p>Paradis, E. (2011) Time-dependent speciation and extinction from phylogenies: a least squares approach. <EM>Evolution</EM>, <B>65</B>, 661&ndash;672. </p>
<p>Paradis, E. (2012) <EM>Analysis of Phylogenetics and Evolution with R (Second Edition).</EM> New York: Springer. </p>
<p>Paradis, E. (2012) Shift in diversification in sister-clade comparisons: a more powerful test. <EM>Evolution</EM>, <B>66</B>, 288&ndash;295. </p>
<p>Paradis, E. and Claude J. (2002) Analysis of comparative data using generalized estimating equations. <EM>Journal of theoretical Biology</EM>, <B>218</B>, 175&ndash;185. </p>
<p>Paradis, E., Claude, J. and Strimmer, K. (2004) APE: analyses of phylogenetics and evolution in R language. <EM>Bioinformatics</EM>, <B>20</B>, 289&ndash;290.
<p>Penny, D. and Hendy, M. D. (1985) The use of tree comparison metrics. <EM>Systemetic Zoology</EM>, <B>34</B>, 75&ndash;82. </p>
<p>Posada, D. and Crandall, K. A. (2001) Selecting the best-fit model of nucleotide substitution. <EM>Systematic Biology</EM>, <B>50</B>, 580&ndash;601. </p>
<p>Purvis, A. and Rambaut, A. (1995) Comparative analysis by independent contrasts (CAIC): an Apple Macintosh application for analysing comparative data. <EM>CABIOS</EM>, <B>11</B> :241&ndash;251. </p>
<p>Pybus, O. G. and Harvey, P. H. (2000) Testing macro-evolutionary models using incomplete molecular phylogenies. <EM>Proceedings of the Royal Society of London. Series B. Biological Sciences</EM>, <B>267</B>, 2267&ndash;2272. </p>
<p>Pybus, O. G, Rambaut, A. and Harvey, P. H. (2000) An integrated framework for the inference of viral population history from reconstructed genealogies. <EM>Genetics</EM>, <B>155</B>, 1429&ndash;1437. </p>
<p>Rabosky, D. L., Donnellan, S. C., Talaba, A. L. and Lovette, I. J. (2007) Exceptional among-lineage variation in diversification rates during the radiation of Australia's most diverse vertebrate clade. <EM>Proceedings of the Royal Society of London. Series B. Biological Sciences</EM>, <B>274</B>, 2915&ndash;2923. </p>
<p>Rohlf, F. J. (2001) Comparative methods for the analysis of continuous variables: geometric interpretations. <EM>Evolution</EM>, <B>55</B>, 2143&ndash;2160. </p>
<p>Rokas, A., Nylander, J. A. A., Ronquist, F. and Stone, G. N. (2002) A maximum likelihood analysis of eight phylogenetic markers in Gallwasps (Hymenoptera: Cynipidae): implications for insect phylogenetic studies. <EM>Molecular Phylogenetics and Evolution</EM>, <B>22</B>, 206&ndash;219.
<p>Rzhetsky, A. and Nei, M. (1992) A simple method for estimating and testing minimum-evolution trees. <EM>Molecular Biology and Evolution</EM>, <B>9</B>, 945&ndash;967. </p>
<p>Saitou, N. and Nei, M. (1987) The neighbor-joining method: a new method for reconstructing phylogenetic trees. <EM>Molecular Biology and Evolution</EM>, <B>4</B>, 406&ndash;425. </p>
<p>Sanderson, M. J. (1997) A nonparametric approach to estimating divergence times in the absence of rate constancy. <EM>Molecular Biology and Evolution</EM>, <B>14</B>, 1218&ndash;1231. </p>
<p>Sanderson, M. J. (2002) Estimating absolute rates of molecular evolution and divergence times: a penalized likelihood approach. <EM>Molecular Biology and Evolution</EM>, <B>19</B>, 101&ndash;109. </p>
<p>Sargent, R. D. (2004) Floral symmetry affects speciation rates in angiosperms. <EM>Proceedings of the Royal Society of London. Series B. Biological Sciences</EM>, <B>271</B>, 603&ndash;608. </p>
<p>Schluter, D., Price, T., Mooers, A. O. and Ludwig, D. (1997) Likelihood of ancestor states in adaptive radiation. <EM>Evolution</EM>, <B>51</B>, 1699&ndash;1711. </p>
<p>Siegel, S. and Castellan, N. J., Jr. (1988) <EM>Nonparametric statistics for the behavioral sciences. 2nd edition</EM>. New York: McGraw-Hill. </p>
<p>Slowinski, J. B. and Guyer, C. (1993) Testing whether certain traits have caused amplified diversification: an improved method based on a model of random speciation and extinction. <EM>American Naturalist</EM>, <B>142</B>, 1019&ndash;1024. </p>
<p>Sokal, R. R. and Rohlf, F. J. (1995) <EM>Biometry - The principles and practice of statistics in biological research. Third edition.</EM> New York: W. H. Freeman. </p>
<p>Stephens, M. A. (1974) EDF statistics for goodness of fit and some comparisons. <EM>Journal of the American Statistical Association</EM>, <B>69</B>, 730&ndash;737. </p>
<p>Strimmer, K. and Pybus, O. G. (2001) Exploring the demographic history of DNA sequences using the generalized skyline plot. <EM>Molecular Biology and Evolution</EM>, <B>18</B>, 2298&ndash;2305. </p>
<p>Studier, J. A. and Keppler, K. J. (1988) A note on the neighbor-joining algorithm of Saitou and Nei. <EM>Molecular Biology and Evolution</EM>, <B>5</B>, 729&ndash;731. </p>
<p>Swofford, D. L. and Maddison, W. P. (1987) Reconstructing ancestral character states under Wagner parsimony. <EM>Mathematical Biosciences</EM>, <B>87</B>, 199&ndash;229. </p>
<p>Tamura, K. (1992) Estimation of the number of nucleotide substitutions when there are strong transition-transversion and G + C-content biases. <EM>Molecular Biology and Evolution</EM>, <B>9</B>, 678&ndash;687. </p>
<p>Tamura, K. and Nei, M. (1993) Estimation of the number of nucleotide substitutions in the control region of mitochondrial DNA in humans and chimpanzees. <EM>Molecular Biology and Evolution</EM>, <B>10</B>, 512&ndash;526. </p>
<p>Vamosi, S. M. (2007) Endless tests: guidelines for analysing non-nested sister-group comparisons. An addendum. <EM>Evolutionary Ecology Research</EM>, <B>9</B>, 717. </p>
<p>Vamosi, S. M. and Vamosi, J. C. (2005) Endless tests: guidelines for analysing non-nested sister-group comparisons. <EM>Evolutionary Ecology Research</EM>, <B>7</B>, 567&ndash;579. </p>
<p>Venables, W. N. and Ripley, B. D. (2002) <EM>Modern applied statistics with S (fourth edition)</EM>. New York: Springer-Verlag. </p>
<p>Wiegmann, B., Mitter, C. and Farrell, B. 1993. Diversification of carnivorous parasitic insects: extraordinary radiation or specialized dead end? <EM>American Naturalist</EM>, <B>142</B>, 737&ndash;754. </p>
<hr>
Last updated: Fri Apr 27 07:44:45 2012
</body></html>